src/Analyzers/CSharp/CodeFixes/RemoveUnusedParametersAndValues/CSharpRemoveUnusedValuesCodeFixProvider.cs

```diff
@@ -164,5 +166,32 @@ protected override void InsertAtStartOfSwitchCaseBlockForDeclarationInCaseLabelO
                 return SyntaxFactory.BinaryExpression(mappedBinaryExpressionKind, leftOfAssignment, rightOfAssignment);
             }
         }
+
+        protected override SyntaxNode GetReplacementNodeForVarPattern(SyntaxNode originalVarPattern, SyntaxNode newNameNode)
+        {
+            if (originalVarPattern is not VarPatternSyntax pattern)
+            {
+                Debug.Fail($"The given {nameof(originalVarPattern)} node is not a var pattern, but {originalVarPattern.Kind()}");
```

Comment:
Seems like this, and the one at the end of the method, should be `throw ExceptionUtilities.Unreachable`. A debug fail is most likely only going to show up when running tests, and I assume you haven't written any tests that hit this line, nor would any existing tests.

Reply:
> you haven't written any tests that hit this line, nor would any existing tests.

I don't think it is possible. Unreachable exception indeed seems to better here